debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -336,6 +338,21 @@ protected void execute() {
                         logger.warn("\t skipping database '{}' due to error reading tables: {}", dbName, e.getMessage());
                     }
                 }
+
+                /* To achieve an ordered snapshot, we would first get a list of Regex tables.whitelist regex patterns
+                   and then sort the tableIds list based on the above list
+                 */
+                List<Pattern> tableWhitelistPattern = Strings.listOfRegex(context.config().getString(MySqlConnectorConfig.TABLE_WHITELIST),Pattern.CASE_INSENSITIVE);
+                List<TableId> tableIdsSorted = new ArrayList<>();
+                tableWhitelistPattern.forEach(pattern -> {
+                    List<TableId> tablesMatchedByPattern = tableIds.stream().filter(t -> pattern.asPredicate().test(t.toString()))
+                            .collect(Collectors.toList());
+                    tablesMatchedByPattern.forEach(t -> {
+                        if (!tableIdsSorted.contains(t))
+                            tableIdsSorted.add(t);
+                    });
+                });
+                tableIds.sort(Comparator.comparing(tableIdsSorted::indexOf));
```

Comment:
What is the difference in contents of `tableIds` and `tableIdsSorted`? If only order then maybe plain re-assignment would be enough? If there is a real difference like every list ocntains different elements then the result of this operation would be placing of ordered tables at the end of list. Is it ok?

Reply:
no there's no difference. tableIdsSorted just contains tableIds sorted from the TableIds list in the order retrieved from the whitelist list. So re assigning would be fine. Would make the change.